qa/python_models/response_sender/model_common.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import asyncio
+import threading
+import time
+
+import numpy as np
+
+
+class ResponseSenderModelCommon:
+    def __init__(self, pb_utils):
+        self._pb_utils = pb_utils
+        self._background_tasks = set()
+
+    def _get_instructions_from_request(self, request):
+        instr = {}
+        return_a_response_np = self._pb_utils.get_input_tensor_by_name(
+            request, "RETURN_A_RESPONSE"
+        ).as_numpy()
+        instr["batch_size"] = return_a_response_np.shape[0]
+        instr["return_a_response"] = bool(return_a_response_np.sum())
```

Comment:
Is the idea behind `bool(...sum())` that even if one of the values in the batch is larger than 0, we would consider it true?

Reply:
Yes, because if there is a batched tensor, there is only one output slot for the tensor on the list that is returned. I think for the purpose of this test, we only need to test if a return has taken place or not on a particular slot, and the content of the returned tensor does not matter, so I basically just picked the simplest way of reducing a list of `bool` into one `bool` - if there is one or more true, we consider it true; if all false, we consider it false.

Reply:
sounds good, thanks for the explanation.